staging/src/k8s.io/pod-security-admission/policy/check_seccomp_restricted.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/pod-security-admission/api"
+)
+
+func init() {
+	addCheck(CheckSeccompRestricted)
+}
+
+func CheckSeccompRestricted() Check {
+	return Check{
+		ID:    "seccomp_restricted",
+		Level: api.LevelRestricted,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       seccomp_1_0_restricted,
+			},
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 19),
+				CheckPod:       seccomp_1_19_restricted,
+			},
+		},
+	}
+}
+
+// seccomp_1_0_restricted checks restricted policy on seccomp alpha annotation
+func seccomp_1_0_restricted(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
```

Comment:
The more I think about it, the less sure I am about requiring pods to set alpha annotations in order to comply with the restricted policy < 1.19. Forbidding explicitly choosing "unconfined" is reasonable (the baseline check does that), but I don't think a user should ever be required to add an alpha field or annotation to their pod spec to make it pass pod security checks.

@tallclair, what would you think of making the restricted "must specify seccomp level" check 1.19+ only and only applying to the field?

Reply:
Talked with Tim offline. He suggested that the history of the seccomp feature was a bit weird, and even though the annotation had alpha in the name, it existed prior to feature gates and he thought most users that want the restricted profile would also want to require require seccomp.

I think that use of pinned PodSecurity restricted levels prior to 1.19/1.20 will be very rare... once 1.22 is released, 1.20 will be the oldest supported version. I think it's better to set the pattern that the restricted profile does not require setting fields/annotations/whatever until they are GA.

Tim agreed that use of those older restricted policy levels was likely to be rare, and I convinced him it was reasonable to set up this check as we would want future restricted checks to be done.

So... the adjustment to this PR is to drop the 1.0 annotation-based restricted check/fixture, leaving the 1.19 field-based restricted check/fixture in place and both baseline checks in place